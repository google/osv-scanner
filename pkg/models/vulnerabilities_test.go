package models_test

import (
	"testing"

	"github.com/google/osv-scanner/internal/testutility"
	"github.com/google/osv-scanner/pkg/models"
)

func TestVulnerabilities_MarshalJSON(t *testing.T) {
	t.Parallel()

	tests := []struct {
		name string
		vs   models.Vulnerabilities
	}{
		{
			name: "nil",
			vs:   nil,
		},
		{
			name: "no vulnerabilities",
			vs:   models.Vulnerabilities{},
		},
		{
			name: "one vulnerability",
			vs:   models.Vulnerabilities{models.Vulnerability{ID: "GHSA-1"}},
		},
		{
			name: "multiple vulnerabilities",
			vs: models.Vulnerabilities{
				models.Vulnerability{ID: "GHSA-1"},
				models.Vulnerability{ID: "GHSA-2"},
				models.Vulnerability{ID: "GHSA-3"},
			},
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			t.Parallel()
			testutility.NewSnapshot().MatchJSON(t, tt.vs)
		})
	}
}
