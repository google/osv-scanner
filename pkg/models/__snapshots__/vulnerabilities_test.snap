
[TestVulnerabilities_MarshalJSON/multiple_vulnerabilities - 1]
[
  {
    "modified": "0001-01-01T00:00:00Z",
    "id": "GHSA-1"
  },
  {
    "modified": "0001-01-01T00:00:00Z",
    "id": "GHSA-2"
  },
  {
    "modified": "0001-01-01T00:00:00Z",
    "id": "GHSA-3"
  }
]
---

[TestVulnerabilities_MarshalJSON/nil - 1]
[]
---

[TestVulnerabilities_MarshalJSON/no_vulnerabilities - 1]
[]
---

[TestVulnerabilities_MarshalJSON/one_vulnerability - 1]
[
  {
    "modified": "0001-01-01T00:00:00Z",
    "id": "GHSA-1"
  }
]
---
